proc/proc.go

```diff
@@ -46,6 +46,7 @@ type Process struct {
 	packageMap map[string]string
 
 	allGCache               []*G
+	dwarfLock               sync.Mutex
```

Comment:
Why not add a `Process`-wide lock and let `Debugger` acquire and release it? That would solve this problem and also make the API thread safe: no function call works while `Continue` is running anyway, besides `RequestManualStop`, right?